packages/compass-connections-navigation/src/tree-data.ts

```diff
@@ -0,0 +1,289 @@
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
+import {
+  MAX_COLLECTION_PLACEHOLDER_ITEMS,
+  MAX_DATABASE_PLACEHOLDER_ITEMS,
+  MIN_DATABASE_PLACEHOLDER_ITEMS,
+} from './constants';
+import type {
+  VirtualPlaceholderItem,
+  VirtualTreeItem,
+} from './virtual-list/use-virtual-navigation-tree';
+
+type Collection = {
+  _id: string;
+  name: string;
+  type: string;
+};
+
+type Status = 'initial' | 'fetching' | 'refreshing' | 'ready' | 'error';
+
+type Database = {
+  _id: string;
+  name: string;
+  collectionsStatus: Status;
+  collectionsLength: number;
+  collections: Collection[];
+};
+
+export type Connection = {
+  connectionInfo: ConnectionInfo;
+  name: string;
+  databasesStatus: Status;
+  databasesLength: number;
+  databases: Database[];
+  isReady: boolean;
+  isDataLake: boolean;
+  isWritable: boolean;
+  isPerformanceTabSupported: boolean;
+};
+
+type PlaceholderTreeItem = VirtualPlaceholderItem & {
+  colorCode?: string;
+  id: string;
+  maxNestingLevel: number;
```

Comment:
Feel free to keep, but this belongs more in the virtual tree item data not in the item itself, tree item data is specifically a mechanism that allows us to pass shared values to all items.

Reply:
good idea. will make this change.